### DISCUSSION TITLE
画像が画面上にあるか判定したい

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#52] [Question-original]
１　敵が出ている間は、〜秒にHPが10ずつ減るというコードを書きたいのですが、「〜秒にHPが10ずつ減る」はできたのですが、敵が出ているか出ていないかを判定するコードを書きたいです。

２　if EnemyImage.isHidden=false
のように書けば、敵がもし隠れていない時というふうになると考えたのですが、エラーが出てしまいました。

３　探してもそれに関する記事はありませんでした..

４　if EnemyImage.isHidden=false

５　https://github.com/ShunHigu/Original

**Comment by Yonpachi48** [comment]
一度最新版をpushしていただけると！
それと、`EnemyImage.isHidden=false`はEnemyImage.isHiddenにfalseを代入する(EnemyImageを表示する)というコードで、もしEnemyImageが表示されているかどうかの判定をしたいのなら`if EnemyImage.isHidden==false`になります！

**Reply by ShunHigu** [reply]
ありがとうございます！
そのコードを書く場所は、いきなりifから書き始めるのではなく、
「 override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?)」
のように書かなければいけないと考えたのですが、どのように書くのが適切なのでしょうか？

**Reply by Yonpachi48** [reply]
〜秒にHPが10ずつ減るという仕様ならTimerを使った処理になるかな？
テクモンの教科書の21Pあたりを参考にするといいと思います！
